code/modules/hippie/pool/pool_drain.dm

```diff
@@ -0,0 +1,152 @@
+/obj/machinery/drain
+	name = "Drain"
+	icon = 'icons/turf/pool.dmi'
+	icon_state = "drain"
+	desc = "This removes things that clog the pool."
+	anchored = 1
+	var/active = 0
+	var/status = 0 //1 is drained, 0 is full.
+	var/timer = 0
+	var/cooldown
+	var/obj/machinery/poolcontroller/poolcontrol = null
+
+/obj/machinery/drain/Initialize()
+	for(var/obj/machinery/poolcontroller/control in range(5,src))
+		src.poolcontrol += control
+
+/obj/machinery/drain/process()
+	if(!status) //don't drain an empty pool.
+		for(var/obj/item/absorbo in orange(1,src))
+			if(absorbo.w_class == 1)
+				step_towards(absorbo, src)
+		spawn(7) //Gives them just the time to pick their items up.
+			for(var/obj/item/absorb in range(0,src))
+				if(absorb.w_class == 1)
+					for(var/obj/machinery/poolfilter/filter in range(3,src))
+						absorb.loc = filter
+	if(active)
+		if(status) //if filling up, get back to normal position
+			if(timer > 0)
+				playsound(src, 'sound/effects/fillingwatter.ogg', 100, 1)
+				timer--
+				for(var/obj/whirlo in orange(1,src))
+					if(!whirlo.anchored )
+						step_away(whirlo,src)
+				for(var/mob/living/carbon/human/whirlm in orange(2,src))
+					step_away(whirlm,src)
+			else if(timer == 0)
+				for(var/turf/open/pool/water/undrained in range(5,src))
+					undrained.name = "poolwater"
+					undrained.desc = "You're safer here than in the deep."
+					undrained.icon_state = "turf"
+					undrained.drained = 0
+				for(var/obj/overlay/water/undrained2 in range(5,src))
+					undrained2.icon_state = "overlay"
+				for(var/obj/effect/effect/waterspout/undrained3 in range(1,src))
+					qdel(undrained3)
+				poolcontrol.drained = 0
+				if(poolcontrol.bloody < 1000)
+					poolcontrol.bloody /= 2
+				if(poolcontrol.bloody > 1000)
+					poolcontrol.bloody /= 4
+				poolcontrol.changecolor()
+				status = 0
+				active = 0
+			return
+		if(!status) //if draining, change everything.
+			if(timer > 0)
+				playsound(src, 'sound/effects/pooldrain.ogg', 100, 1)
```

Comment:
Please put these sounds somewhere on the machine itself so they can be changed trough VV